src/libecalc/common/graph.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
```

Comment:
Important that we have good and covered tests for this structure, since it will be the backbone of eCalc